tools/cloudflow-operator/src/main/scala/cloudflow/operator/Operator.scala

```diff
@@ -0,0 +1,297 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cloudflow.operator
+
+import akka.NotUsed
+import akka.actor._
+import akka.datap.crd.App
+import akka.kube.actions.{ Action, ActionExecutor, Fabric8ActionExecutor }
+import akka.stream._
+import akka.stream.scaladsl._
+import cloudflow.operator.action._
+import cloudflow.operator.action.runner.Runner
+import cloudflow.operator.event._
+import cloudflow.operator.flow._
+import io.fabric8.kubernetes.api.model.{ WatchEvent => _, _ }
+import io.fabric8.kubernetes.client.KubernetesClient
+import io.fabric8.kubernetes.client.dsl.base.OperationContext
+import io.fabric8.kubernetes.client.informers.{
+  EventType,
+  ResourceEventHandler,
+  SharedIndexInformer,
+  SharedInformerFactory
+}
+import org.slf4j.LoggerFactory
+
+import java.util.concurrent.Executors
+import java.util.concurrent.atomic.AtomicReference
+import scala.concurrent._
+import scala.concurrent.duration._
+import scala.jdk.CollectionConverters._
+import scala.util._
+
+object Operator {
+  lazy val log = LoggerFactory.getLogger("Operator")
+
+  val ProtocolVersion = "5"
+  val ProtocolVersionKey = "protocol-version"
+  val ProtocolVersionConfigMapName = "cloudflow-protocol-version"
+  def ProtocolVersionConfigMap(ownerReferences: List[OwnerReference]) = {
+    new ConfigMapBuilder()
+      .withNewMetadata()
+      .withName(ProtocolVersionConfigMapName)
+      .withLabels((Map(ProtocolVersionConfigMapName -> ProtocolVersionConfigMapName)).asJava)
+      .withOwnerReferences(ownerReferences: _*)
+      .endMetadata()
+      .withData(Map(ProtocolVersionKey -> ProtocolVersion).asJava)
+      .build()
+  }
+
+  val AppIdLabel = "com.lightbend.cloudflow/app-id"
+  val ConfigFormatLabel = "com.lightbend.cloudflow/config-format"
+  val StreamletNameLabel = "com.lightbend.cloudflow/streamlet-name"
+  val ConfigUpdateLabel = "com.lightbend.cloudflow/config-update"
+
+  val DefaultWatchOptions = Map(CloudflowLabels.ManagedBy -> CloudflowLabels.ManagedByCloudflow)
+
+  val MaxObjectBufSize = 8 * 1024 * 1024
+
+  val restartSettings = RestartSettings(minBackoff = 3.seconds, maxBackoff = 30.seconds, randomFactor = 0.2)
+
+  val decider: Supervision.Decider = {
+    case _ => Supervision.Stop
+  }
+
+  val StreamAttributes = ActorAttributes.supervisionStrategy(decider)
+
+  private lazy val fabric8ExecutionContext = ExecutionContext.fromExecutorService(Executors.newSingleThreadExecutor())
+
+  def handleEvents(client: KubernetesClient, runners: Map[String, Runner[_]], podName: String, podNamespace: String)(
+      implicit system: ActorSystem,
+      mat: Materializer,
+      ec: ExecutionContext) = {
+
+    // handleAppEvents
```

Comment:
why not have separate methods?

Reply:
to share the `sharedInformerFactory` across the streams, we can split it up, but, in case we should push to the `Main` the `informerFactory`, as you prefer ...